pxr/usd/bin/usddiff/usddiff.py

```diff
@@ -170,16 +172,33 @@ def listFiles(dirpath):
                     for file in files]
         return set(ret)
 
-    # Must have FILE FILE, DIR DIR, DIR FILES... or FILES... DIR.
+    origArgs = args
+
+    # path may not be a "real" path due to custom resolvers... so use
+    # Sdf.Layer.FindOrOpen first
```

Comment:
Hmm, good point. Essentially want we want to do here is just validate using the ArResolver. I actually posted a question on the mailing list about this (and some other resolver issues): [https://groups.google.com/forum/#!topic/usd-interest/E3huSatYrLI]https://groups.google.com/forum/#!topic/usd-interest/E3huSatYrLI

Assuming that we can rely on getting back an empty string as a form of validation, I think maybe the best thing to do is to just replace the look that checks for existence with one which checks that either it resolves to non-empty, OR it's a dir (per your next note).